executor/cook/executor.py

```diff
@@ -199,7 +244,7 @@ def kill_task(process, shutdown_grace_period_ms):
                 break
         if is_process_running(process):
             logging.info('Process did not terminate, forcefully killing it')
-            process.kill()
+            kill_process_group(group_id, signal.SIGKILL)
```

Comment:
Nitpick: I think it might make more sense to eliminate this branch altogether. I think that in almost all cases the shell that we use to launch the command is going to exit due to the `SIGTERM`, and `is_process_running` is only checking the status of that process. It might make more sense to always assume that the `SIGKILL` in the `finally` block is going to clean up any processes that caught the `SIGTERM`.

On another note, it might make sense for us to encourage using `exec` in the Executor command so that the wrapper shell in the launched process is actually *replaced* by the main executor app. Using `exec` like that would cause these `process.poll()` checks to check the process that we actually care about.